src/pages/NewUser/helper.tsx

```diff
@@ -56,28 +68,33 @@ export const newUserFormProps: FormProps = {
 					label: 'Admin',
 				},
 				{
-					value: 'user',
-					label: 'User',
+					value: 'garson',
+					label: 'Garson',
 				},
 			],
 		},
 	],
 	button: {
 		text: 'Create',
-		// clickHandler: () => console.log('Create button clicked')
 	},
 }
 
 export const addNewUser = async ({ request }: ActionFunctionArgs) => {
 	let formData = await request.formData()
+	const userId = `${new Date().getTime().toString().split('').reverse().splice(0, 8).join('')}`
 
-	const newUserData: NewUserProps = {
-		name: formData.get('name'),
+	const newUserData: INewUserProps = {
+		firstName: formData.get('first name'),
+		lastName: formData.get('last name'),
 		email: formData.get('email'),
 		password: formData.get('password'),
-		role: formData.get('role'),
+		role: formData.get('role') as ROLES,
+		id: userId,
 	}
-	await createNewUserWittEmailAndPassword(newUserData)
 	await createNewUser(newUserData)
+	if (auth.currentUser) {
+		newUserData.uid = auth.currentUser.uid
```

Comment:
could you please explain what this line actually does? 
looks to me that you assigning the same uid to the new user that current user has...

or is this the piece of code that solves the bug with automatic login after new user is created?

Reply:
![image](https://github.com/user-attachments/assets/e5c56da5-2578-4990-abd5-3138a0b26758)
function createNewUser on row 94 is creating newUser in fireBaseAUTH, after that we have AUTH object where i take UID, this UID i add to newUser object and calls addUserToCollection function, which are creating new user in collection on FB (sinply: i waiting for UID and after expands object by this atribute)